lib/streamlit/elements/select_slider.py

```diff
@@ -147,14 +147,14 @@ def deserialize_select_slider(ui_value, widget_id=""):
                 ui_value = slider_value
 
             # The widget always returns floats, so convert to ints before indexing
-            return_value = list(map(lambda x: options[int(x)], ui_value))  # type: ignore[no-any-return]
+            return_value = list(map(lambda x: opt[int(x)], ui_value))  # type: ignore[no-any-return]
 
             # If the original value was a list/tuple, so will be the output (and vice versa)
             return tuple(return_value) if is_range_value else return_value[0]
 
         def serialize_select_slider(v):
             to_serialize = v if is_range_value else [v]
-            return [index_(options, u) for u in to_serialize]
+            return [index_(opt, u) for u in to_serialize]
```

Comment:
Seeing this. Any reason we don't just use `index_` instead of the `index` method. Might make things easier to reason about with types and access.

Reply:
I didn't even look at the `index_` function. It might be better to use, though I don't know if it does the right thing for a DF or Index. I can look into it though.

Reply:
@kajarenc handled a bug with it, so he might have some added context here.

Reply:
I tried removing the conversion to list that Will added, and instead use `index_` in the multiselect code, and several tests fail due to it not finding the default values in `options`. So it might be useful to use as a supplementary thing, but I think we need to do the initial conversion as well.

Reply:
Sounds good. What's the best way to capture the conversation that needs to happen? I can approve this as it is for now for the sake of patching, but I think this might fall into creating some abstraction for future (and to avoid the disjointed handling of `options`).

Reply:
Probably just needs a ticket with a summary of the issue and a link to this PR for reference?

Reply:
Works for me. Will approve, and we'll get the ticket with the summary in.